SU2_CFD/include/output/CFlowOutput.hpp

```diff
@@ -199,6 +199,70 @@ class CFlowOutput : public CFVMOutput{
    */
   void Set_NearfieldInverseDesign(CSolver *solver, const CGeometry *geometry, const CConfig *config);
 
+  /*!
+   * \brief Compute the custom outputs.
+   * \param[in] solver - The container holding all solution data.
+   * \param[in] geometry - Geometrical definition of the problem.
+   * \param[in] config - Definition of the particular problem.
+   */
+  void SetCustomOutputs(const CSolver *solver, const CGeometry *geometry, const CConfig *config);
+
+  /*!
+   * \brief Helper for custom outputs, converts variable names to indices and pointers which are then used
+   * to evaluate the custom expressions.
+   */
+  template <class FlowIndices>
+  void ConvertVariableSymbolsToIndices(const FlowIndices& idx, CustomOutput& output) const {
+
+    static const auto knownVariables =
+        "TEMPERATURE, TEMPERATURE_VE, VELOCITY_X, VELOCITY_Y, VELOCITY_Z, PRESSURE,\n"
+        "DENSITY, ENTHALPY, SOUND_SPEED, LAMINAR_VISCOSITY, EDDY_VISCOSITY, THERMAL_CONDUCTIVITY";
+
+    auto IndexOfVariable = [&](const FlowIndices& idx, const std::string& var) {
+      if ("TEMPERATURE" == var) return idx.Temperature();
+      if ("TEMPERATURE_VE" == var) return idx.Temperature_ve();
+      if ("VELOCITY_X" == var) return idx.Velocity();
```

Comment:
I think the strategy to handle variables of other solvers would be something like parsing `SCALAR[0], HEAT[0], TURB[0]` into index of the solver + index of variable.